sdk/core/azure-core/src/main/java/com/azure/core/util/Option.java

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util;
+
+import java.util.NoSuchElementException;
+
+/**
+ * The Option type to describe tri-state. Every Option instance is in one of
+ * the three states: a state representing a non-null-value, null-value, or no-value.
+ *
+ * @param <T> The value type.
+ */
+public final class Option<T> {
```

Comment:
Are we going to add a `JsonSerializer<Option>` implementation to `JacksonAdapter`?

Reply:
yes, we will, it will look very close to the serializer for `Optional` in `Java8Module`.

Reply:
Serializer for Option type is added.